types/loadable__component/index.d.ts

```diff
@@ -47,7 +47,20 @@ export interface LoadableComponentMethods<Props> {
     load(props?: Props): Promise<React.ComponentType<Props>>;
 }
 
-export type LoadableComponent<Props> = React.ComponentType<Props & { fallback?: JSX.Element }> &
+export interface ExtraComponentProps {
+    fallback?: JSX.Element;
+}
+
+export type LoadableComponent<Props> = React.ComponentType<Props & ExtraComponentProps> &
+    LoadableComponentMethods<Props>;
+
+export interface ExtraClassComponentProps<Component extends React.ComponentClass> extends ExtraComponentProps {
+    ref?: React.LegacyRef<InstanceType<Component>>;
+}
+
+export type LoadableClassComponent<Props, Component extends React.ComponentClass> = React.ComponentType<
```

Comment:
I don't think two type parameters are necessary here? Seems like there could be just `<Props>` and then you can use `React.ComponentClass<Props>`.

Reply:
Unfortunately that isn't sufficient; the `ref` needs to have a reference to the actual `Component` class type.

If the class has a public method like this:

```tsx
class TestClassComponent extends React.Component<TestProps> {
    publicMethod() {}
    // ...
}
```

..then the ref should be able to call it:

```tsx
    <LoadableClassDirect
        ref={ref => {
            ref && ref.publicMethod();
        }}
        foo=""
    />;
```

Reply:
Right yes - perhaps the other way around then, i.e. keep the `Component` type parameter but remove the separate `Props` parameter, and instead use `React.ComponentProps<Component>` in place of `Props`?